### PR TITLE
GFORMS-2690 - The specimen link for "get-help-to-pay-the-right-tax-as…

### PR DESCRIPTION
**Pull request by DursunGundogan**
…-wealthy-individual" is not functioning correctly and is redirecting to a technical error page.

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/NewFormController.scala [hunk 1]
```diff
@@ -492,9 +492,13 @@ class NewFormController(
     l: LangADT
   ): Future[Result] =
     for {
-      itmpRetrievals <- formTemplate.dataRetrieve.fold(Future.successful(Option.empty[ItmpRetrievals]))(_ =>
-                          auth.getItmpRetrievals(request).map(Some(_))
-                        )
+      itmpRetrievals <- if (formTemplate.isSpecimen) {
+                          Future.successful(Option.empty[ItmpRetrievals])
```
Comment (lxol) [inline app/uk/gov/hmrc/gform/gform/NewFormController.scala:496]: does it work if some fields has includeIf that depends on itmpRetrievals or is it just ignores includeIf when in specimen template mode?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/gform/NewFormController.scala:496]: When posting a template, it removes all includeIfs if the template is a specimen template. It's working as expected.

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/gform/NewFormController.scala:496]: I think we don't need to call  the external APIs for the specimen template.

**Review by lxol — APPROVED** [review]

File: app/uk/gov/hmrc/gform/gform/NewFormController.scala (hunk 1 above)
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/gform/NewFormController.scala:496]: Shouldn't we remove `dataRetrieve` from specimen instead?

**Review by josef-vlach — CHANGES_REQUESTED** [review]
Specimen should not need any special logic.

**Review by josef-vlach — APPROVED** [review]
Hm, removing `dataRetrieve` is not so easy, as there may be a lot of dependencies. Let's go forward with this solution.

**Pull request merged**